### DISCUSSION TITLE
How to point autogen.sh to our own sourced and compiled ffmpeg/libavcodec

### DISCUSSION THREAD

**Discussion by pcarrillom** [CESNET/UltraGrid#179] [Q&A]
Hello,

We are trying to compile ultragrid in UBuntu 18.04, but autogen.sh won’t find our own ffmpeg build. So it is not compiling libavcodec encoders.
Where is autogen looking for libavcodec/ffmpeg?
Which files is exactly autogen looking for?
Can we point autogen to a specific location for this?

Thanks!

**Comment by MartinPulec** [comment]
Hi,

you can look at [this part of configure](https://github.com/CESNET/UltraGrid/blob/2eedee6bbed9be5bb8e17c8128e77d36b6befd2f/configure.ac#L1771-L1789).

First, _pkg-config_ lookup is performed and if libavcodec (et al.) is not found, individual headers/libs are searched for. So I'd suggest to point _PKG\_CONFIG\_PATH_ to your FFmpeg location (the _pkgconfig/_ subdirectory). Otherwise (the second look-up option), you'd need to set _CPATH_/_LIBRARY\_PATH_ (and perhaps also _LD_LIBRARY\_PATH_). For debugging, you can also look to _config.log_ to see what was not found (if something).

Martin
